pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -863,10 +867,25 @@ monitorLoop:
 			l.setMigrationResult(vmi, true, fmt.Sprintf("%v", err), "")
 			break monitorLoop
 		case libvirt.DOMAIN_JOB_CANCELLED:
-			logger.Info("Migration was canceled")
+			logger.Info("Migration job was canceled")
 			l.setMigrationResult(vmi, true, "Live migration aborted ", v1.MigrationAbortSucceeded)
 			break monitorLoop
 		}
+
+		select {
```

Comment:
I'm nervous about moving this. It's unclear if we'll end up in a situation where the monitor loop exits and no result is posted. That would cause the VMI to never be migratable again. Looking at the various `break` calls, I think it is possible and perhaps more likely if we move this code block?

fyi, I spent a bit of time studying the existing logic and attempted to refactor this function in a more consistent way here, #5330.

Reply:
> I'm nervous about moving this. It's unclear if we'll end up in a situation where the monitor loop exits and no result is posted. That would cause the VMI to never be migratable again. Looking at the various `break` calls, I think it is possible and perhaps more likely if we move this code block?

fwiw, I think this shouldn't happen. `break` should only send us to the top of the loop; only `break monitorLoop` will send us out... 




> fyi, I spent a bit of time studying the existing logic and attempted to refactor this function in a more consistent way here, #5330.

yeah and I think it's great, however, my preference would be to fix the bug and then proceed with the refactor.